akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -44,23 +44,31 @@ abstract class HttpApp extends Directives {
   }
 
   /**
-   * Start a server on the specified host and port, using provided settings.
+   * Start a server on the specified host and port, using the provided [[ActorSystem]].
+   * Note that this method is blocking
+   */
+  def startServer(host: String, port: Int, system: ActorSystem): Unit = {
+    startServer(host, port, ServerSettings(system), Some(system))
```

Comment:
To be fair we don't validate many fields for non-nullity, but sure, I can check if it's null and call further with `None`. This should do the trick